download_rpms

```diff
@@ -8,14 +8,20 @@ cd "$RPMDIR"
 tags=""
 
 for os in $OSES; do
-  tags="foreman-$VERSION-$os"
-  [[ -n $ROR_BASE_TAG ]] && tags="$tags $ROR_BASE_TAG-$os"
-  [[ $os =~ rhel ]] && tags="$tags foreman-$VERSION-nonscl-$os"
+  tags="$PROJECT-$VERSION-$os"
+  if [[ $PROJECT = foreman ]]; then
+    [[ -n $ROR_BASE_TAG ]] && tags="$tags $ROR_BASE_TAG-$os"
+    [[ $os =~ rhel ]] && tags="$tags foreman-$VERSION-nonscl-$os"
```

Comment:
I'm wondering if we can have a `HAS_NONSCL=1` variable:
```suggestion
    [[ $HAS_NONSCL == 1 ]] && [[ $os =~ rhel ]] && tags="$tags $PROJECT-$VERSION-nonscl-$os"
```

Alternative implementation:

```bash
# Foreman
ROR_BASE_TAG=...
ADDITIONAL_RHEL_TAG=nonscl
# Katello
ROR_BASE_TAG=""
ADDITIONAL_RHEL_TAG=thirdparty-candlepin
```

Then
```bash
for os in $OSES; do
   tags="$PROJECT-$VERSION-$os" 
   [[ -n $ROR_BASE_TAG ]] && tags="$tags $ROR_BASE_TAG-$os" 
   [[ $os =~ rhel ]] && tags="$tags $PROJECT-$VERSION-$ADDITIONAL_RHEL_TAG-$os" 
done
```

Reply:
Why do we need this complexity anyways? Why not having a `ADDITIONAL_TAGS` (in the version specific settings) and iterate over them?

```bash
# Foreman
ADDITIONAL_TAGS="foreman-rails-$VERSION foreman-$VERSION-nonscl"

# Katello
ADDITIONAL_TAGS="katello-$VERSION-thirdparty-candlepin"

for os in $OSES; do
  tags="$PROJECT-$VERSION-$os"
  for at in ${ADDITIONAL_TAGS}; do
    tags="$tags $at-$os"
  done
done
```

Or even better, non-version-specific:

```bash
# Foreman
ADDITIONAL_TAGS="foreman-rails-@VERSION@ foreman-@VERSION@-nonscl"

# Katello
ADDITIONAL_TAGS="katello-@VERSION@-thirdparty-candlepin"

for os in $OSES; do
  additional_versioned_tags=$(echo ${ADDITIONAL_TAGS} | sed "s/@VERSION@/$VERSION/g")
  tags="$PROJECT-$VERSION-$os"
  for at in ${additional_versioned_tags}; do
    tags="$tags $at-$os"
  done
done
```

Reply:
My issue was that templating part. I didn't think of pulling in sed. That does look like a much cleaner solution. If you're already doing that, wouldn't you want to go all the way and also make the os part of the template and just make it `TAGS`?

Reply:
I made server_tags and client_tags now. it generates:

```
$ ./download_rpms
foreman-1.23-rhel7 foreman-1.23-nonscl-rhel7 foreman-rails-1.23-rhel7 foreman-client-1.23-rhel5 foreman-client-1.23-rhel6 foreman-client-1.23-rhel7 foreman-client-1.23-sles11 foreman-client-1.23-sles12 foreman-client-1.23-fedora28 foreman-client-1.23-fedora29

$ PROJECT=katello VERSION=3.13 ./download_rpms                   
katello-3.13-rhel7 katello-3.13-thirdparty-candlepin-rhel7
```

so looks correct to me?